src/plumpy/rmq/process_control.py

```diff
@@ -0,0 +1,348 @@
+# -*- coding: utf-8 -*-
+"""Module for process level communication functions and classes"""
+
+from __future__ import annotations
+
+import asyncio
+from typing import Any, Dict, Hashable, Optional, Sequence, Union
+
+import kiwipy
+
+from plumpy import loaders
+from plumpy.coordinator import Coordinator
+from plumpy.message import (
+    Intent,
+    MessageBuilder,
+    MessageType,
+    create_continue_body,
+    create_create_body,
+    create_launch_body,
+)
+from plumpy.utils import PID_TYPE
+
+__all__ = [
+    'RemoteProcessController',
+    'RemoteProcessThreadController',
+]
+
+ProcessResult = Any
+ProcessStatus = Any
+
+
+# FIXME: the class not fit typing of ProcessController protocol
```

Comment:
What is the idea here to fix this? Just the class?

Reply:
Okay this was regarding the return type that needs to consider the asyncio future type, that needs to be done in the protocol then it can support the `RemoteProcessThreadController` and `RemoteProcessController`

Reply:
Check in second review if adapted.